apisx/v1alpha1/listenerset_types.go

```diff
@@ -0,0 +1,580 @@
+	//
+	// +kubebuilder:validation:MaxItems=8
+	SupportedKinds []RouteGroupKind `json:"supportedKinds"`
+
+	// AttachedRoutes represents the total number of Routes that have been
+	// successfully attached to this Listener.
+	//
+	// Successful attachment of a Route to a Listener is based solely on the
+	// combination of the AllowedRoutes field on the corresponding Listener
+	// and the Route's ParentRefs field. A Route is successfully attached to
+	// a Listener when it is selected by the Listener's AllowedRoutes field
+	// AND the Route has a valid ParentRef selecting the whole Gateway
+	// resource or a specific Listener as a parent resource (more detail on
+	// attachment semantics can be found in the documentation on the various
+	// Route kinds ParentRefs fields). Listener or Route status does not impact
+	// successful attachment, i.e. the AttachedRoutes field count MUST be set
+	// for Listeners with condition Accepted: false and MUST count successfully
+	// attached Routes that may themselves have Accepted: false conditions.
+	//
+	// Uses for this field include troubleshooting Route attachment and
+	// measuring blast radius/impact of changes to a Listener.
+	AttachedRoutes int32 `json:"attachedRoutes"`
+
+	// Conditions describe the current condition of this listener.
+	//
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+// ListenerSetConditionType is a type of condition associated with a
+// ListenerSet. This type should be used with the ListenerSetStatus.Conditions
+// field.
+type ListenerSetConditionType string
+
+// ListenerSetConditionReason defines the set of reasons that explain why a
+// particular ListenerSet condition type has been raised.
+type ListenerSetConditionReason string
+
+const (
+	// This condition indicates whether a ListenerSet has generated some
+	// configuration that is assumed to be ready soon in the underlying data
+	// plane.
+	//
+	// It is a positive-polarity summary condition, and so should always be
+	// present on the resource with ObservedGeneration set.
+	//
+	// It should be set to Unknown if the controller performs updates to the
+	// status before it has all the information it needs to be able to determine
+	// if the condition is true.
+	//
+	// Possible reasons for this condition to be True are:
+	//
+	// * "Programmed"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	//
+	// Additional reasons for this condition to be False are influenced by
+	// child ListenerEntry conditions:
+	//
+	// * "PortUnavailable"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ListenerSetConditionProgrammed ListenerSetConditionType = "Programmed"
```

Comment:
In what cases would we expect this status to be different than the parent status? Could a Gateway be "Programmed" but not a ListenerSet? What about the other way around?

Reply:
I would expect that a ListenerSet cannot be programmed in a non-programmed Gateway, while the other way around would be possible.

Reply:
yeah I agree with @mlavacca

Reply:
let me add a `GatewayNotProgrammed` Reason